airflow/utils/log/task_context_logger.py

```diff
@@ -78,7 +109,10 @@ def _get_task_handler() -> FileTaskHandler | None:
             assert isinstance(h, FileTaskHandler)
         return h
 
-    def _log(self, level: int, msg: str, *args, ti: TaskInstance):
+    @provide_session
+    def _log(
```

Comment:
As a consequence, this will create a new `Session` every-time this function is called (every-time we log)

Reply:
Interesting, that could be a lot. I wonder if you should defer the session creation until you know that you got a ti key instead of an actual ti (inside ensure_ti), that would minimize the sessions created to only when you need it at least.